source/views/sis/student-work.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+/**
+ * All About Olaf
+ * Student Work page
+ */
+
+import React from 'react'
+import {StyleSheet, Platform, Text} from 'react-native'
+import * as c from '../components/colors'
+import SimpleListView from '../components/listview'
+import {ListSeparator, ListSectionHeader} from '../components/list'
+import {tracker} from '../../analytics'
+import bugsnag from '../../bugsnag'
+import {NoticeView} from '../components/notice'
+import LoadingView from '../components/loading'
+import delay from 'delay'
+import size from 'lodash/size'
+import sortBy from 'lodash/sortBy'
+import groupBy from 'lodash/groupBy'
+import {JobRow} from './student-work/job-row'
+import type {JobType} from './student-work/types'
+
+const jobsUrl =
+  'https://www.stolaf.edu/apps/stuwork/index.cfm?fuseaction=getall&nostructure=1'
+
+const headerHeight = Platform.OS === 'ios' ? 33 : 41
+
+const jobSort = new Map([
+  ['On-campus Work Study', 1],
+  ['Off-campus Community Service Work Study', 2],
+  ['On-campus Summer Employment', 3],
+  ['Off-campus Summer Employment', 4],
+])
+
+const styles = StyleSheet.create({
+  listContainer: {
+    backgroundColor: c.white,
+  },
+  rowSectionHeader: {
+    height: headerHeight,
+  },
+})
+
+export default class StudentWorkView extends React.Component {
+  state: {
+    jobs: {[key: string]: JobType[]},
+    loading: boolean,
+    refreshing: boolean,
+    error: boolean,
+  } = {
+    jobs: {},
+    loading: false,
+    refreshing: false,
+    error: false,
+  }
+
+  componentWillMount() {
+    this.refresh()
+  }
+
+  fetchData = async () => {
+    try {
+      const data: {[key: string]: JobType[]} = await fetchJson(jobsUrl)
+
+      const sorted = sortBy(data, [
+        j => jobSort.get(j.type) || Infinity,
```

Comment:
Add a comment, like `// We have predefined orders for some job types, but we want all unknown types to show up at the end of the list` (but wrapped to 80 columns)